src/api/app/controllers/status/required_checks_controller.rb

```diff
@@ -0,0 +1,98 @@
+class Status::RequiredChecksController < ApplicationController
+  #### Includes and extends
+
+  #### Constants
+
+  #### Self config
+
+  #### Callbacks macros: before_action, after_action, etc.
+  before_action :set_checkable, only: [:index, :create, :destroy]
+  before_action :set_required_check, only: [:destroy]
+  skip_before_action :require_login, only: [:index]
+  # Pundit authorization policies control
+  after_action :verify_authorized
+
+  #### CRUD actions
+
+  # GET /projects/:project_name/repositories/:repository_name/required_checks
+  def index
+    authorize @checkable
+    @required_checks = @checkable.required_checks
+  end
+
+  # POST /projects/:project_name/repositories/:repository_name/required_checks
+  def create
+    authorize @checkable
+    @required_checks = @checkable.required_checks |= names
+
+    if @checkable.save
+      render action: :index
+    else
+      render_error(
+        status: 422,
+        errorcode: 'invalid_required_check',
+        message: "Could not save required check: #{@checkable.errors.full_messages.to_sentence}"
+      )
+    end
+  end
+
+  # DELETE /projects/:project_name/repositories/:repository_name/required_checks/:name
+  def destroy
+    authorize @checkable
+    @checkable.required_checks -= [params[:name]]
+
+    if @checkable.save
+      render_ok
+    else
+      render_error(
+        status: 422,
+        errorcode: 'invalid_required_check',
+        message: "Could not delete required check: #{@checkable.errors.full_messages.to_sentence}"
+      )
+    end
+  end
+
+  #### Non CRUD actions
+
+  #### Non actions methods
+  # Use hide_action if they are not private
+
+  private
+
+  def set_checkable
+    @project = Project.get_by_name(params[:project_name])
```

Comment:
Doing this when `params[:repository_name]` is `nil` is unnecessary. I would move it in the `if` on the line below.

Reply:
Moved down although we need to move it up as soon as we support project or package required checks ...